plugins/modules/ovirt_disk.py

```diff
@@ -420,118 +428,175 @@ def create_transfer_connection(module, transfer, context, connect_timeout=10, re
     return connection, url
 
 
-def transfer(connection, module, direction, transfer_func):
+def start_transfer(connection, module, direction):
     transfers_service = connection.system_service().image_transfers_service()
+    hosts_service = connection.system_service().hosts_service()
     transfer = transfers_service.add(
         otypes.ImageTransfer(
-            image=otypes.Image(
-                id=module.params['id'],
-            ),
+            disk=otypes.Disk(id=module.params.get('id')),
             direction=direction,
+            timeout_policy=otypes.ImageTransferTimeoutPolicy.LEGACY,
+            host=otypes.Host(
+                id=get_id_by_name(hosts_service, module.params.get('host'))
+            ) if module.params.get('host') else None,
+            # format=raw uses the NBD backend, enabling:
+            # - Transfer raw guest data, regardless of the disk format.
+            # - Automatic format conversion to remote disk format. For example,
+            #   upload qcow2 image to raw disk, or raw image to qcow2 disk.
+            # - Collapsed qcow2 chains to single raw file.
+            # - Extents reporting for qcow2 images and raw images on file storage,
+            #   speeding up downloads.
+            format=otypes.DiskFormat.RAW,
         )
     )
     transfer_service = transfers_service.image_transfer_service(transfer.id)
 
-    try:
-        # After adding a new transfer for the disk, the transfer's status will be INITIALIZING.
-        # Wait until the init phase is over. The actual transfer can start when its status is "Transferring".
-        while transfer.phase == otypes.ImageTransferPhase.INITIALIZING:
-            time.sleep(module.params['poll_interval'])
-            transfer = transfer_service.get()
+    start = time.time()
 
-        context = ssl.create_default_context()
-        auth = module.params['auth']
-        if auth.get('insecure'):
-            context.check_hostname = False
-            context.verify_mode = ssl.CERT_NONE
-        elif auth.get('ca_file'):
-            context.load_verify_locations(cafile=auth.get('ca_file'))
-
-        transfer_connection, transfer_url = create_transfer_connection(module, transfer, context)
-        transfer_func(
-            transfer_service,
-            transfer_connection,
-            transfer_url,
-        )
-        return True
-    finally:
-        transfer_service.finalize()
-        while transfer.phase in [
-            otypes.ImageTransferPhase.TRANSFERRING,
-            otypes.ImageTransferPhase.FINALIZING_SUCCESS,
-        ]:
-            time.sleep(module.params['poll_interval'])
+    while True:
```

Comment:
The code was copied as-is from the SDK.
As discussed with Martin Necas I will investigate the issue more deeply there and if any change/improvement will be done in the SDK the same change will be copied to the Ansible Collection project.
So I'll give my "+1" to unblock this commit for merging :)
I do think better remove lines 469-471 (if transfer.phase == otypes.ImageTransferPhase.CANCELLED_SYSTEM) to keep the code the same as in SDK though. So it will be easier to propagate changes/fixes from SDK to here.

Reply:
Can be marked as resolved :)